File: Floyd.cpp
#include <iostream>
using namespace std;
int minshortpath[15][15];

void algorithm(int original[15][15], int minshortpath[15][15]){
  int i,j,k;
  for(k=0;k<15;k++){
    for(i=0;i<15;i++){
      for(j=0;j<15;j++){
        if(original[i][k]+ original[k][j] < original[i][j]){
          original[i][j] = original[i][k] + original[k][j];
          minshortpath[i][j] = k;
        }
      }
    }
  }
}

void shortest(int i, int j){
  int k = minshortpath[i][j];
  if (k > 0){
    shortest(i,k);
    cout<<"->"<<k<<"->";
    shortest(k,j);
  }
}

void findpath(int original[15][15], int i, int j)
{
  cout<<"Path from " << i <<" to "<< j << ": ";
  if (original[i][j]  < 6000)
    {
      cout<<i;
      shortest(i, j);
      cout<<"->"<<j<<"->";
    }
  }

int main(){

int original[15][15] =
{
  {0,5000,5000,663,5000,5000,5000,5000,5000,475,5000,5000,5000,5000,5000},
  {5000,0,5000,5000,5000,1780,1423,5000,5000,948,5000,5000,5000,2060,5000},
  {5000,5000,0,5000,1440,5000,5000,949,5000,517,1614,5000,5000,5000,5000},
  {663,5000,5000,0,5000,5000,5000,5000,811,5000,237,5000,5000,5000,5000},
  {5000,5000,1440,5000,0,272,5000,5000,5000,5000,5000,5000,5000,5000,414},
  {5000,1780,5000,5000,272,0,5000,5000,5000,5000,5000,5000,5000,5000,5000},
  {5000,1423,5000,5000,5000,5000,0,5000,5000,5000,5000,5000,5000,5000,5000},
  {5000,5000,949,5000,5000,5000,5000,0,5000,5000,1217,792,5000,5000,5000},
  {5000,5000,5000,811,5000,5000,5000,5000,0,5000,5000,5000,1771,5000,2257},
  {475,948,517,5000,5000,5000,5000,5000,5000,0,5000,5000,5000,5000,5000},
  {5000,5000,1614,237,5000,5000,5000,1217,5000,5000,0,5000,5000,5000,5000},
  {5000,5000,5000,5000,5000,5000,5000,792,5000,5000,5000,0,5000,5000,5000},
  {5000,5000,5000,5000,5000,5000,5000,1771,5000,5000,5000,5000,0,5000,5000},
  {5000,2060,5000,5000,5000,5000,5000,5000,5000,5000,5000,5000,5000,0,804},
  {5000,5000,5000,5000,414,5000,5000,5000,2257,5000,5000,5000,5000,804,0}
};

cout << "Original Matrix: " << endl;
for (int i = 0; i < 15; i++){
  for(int x = 0; x < 15; x++){
    cout << original[i][x] << ' ';
  }
  cout << endl;
}
cout << endl;

algorithm(original, minshortpath);
for(int i = 0; i<=15; i++){
  for(int j = 0; j<15; j++){
    if(i==j){
      cout << "Path from " << i << " to " << j << ": 0 " << endl;
    }
    else if (i != j){
      findpath(original,i,j);
      cout << j;
      cout << endl;
    }
  }
}
  return 0;
}

File: floyd.py
#!/user/bin/python
#This program implements Floyd's algorithm
#Aisha Jackson
#November 15, 2014

print "This program is going to implement Floyd's Algorithm with the major cities in the United States\n"

print "The matrix is considering the miles between each location\n"

print "The cities are: "
print "1. New York, NY"
print "2. Los Angeles, CA"
print "3. Chicago, IL"
print "4. Houston, TX"
print "5. Philadelphia, PA"
print "6. Phoenix, AZ"
print "7. San Antonio, TX"
print "8. San Diego, CA"
print "9. Dallas, TX"
print "10. San Jose, CA\n"

global startmatrix

startmatrix = (
    [    0,  2792,  790, 1627,   95, 2407, 1822, 2759, 1547, 2941  ], 
    [ 2792,     0, 2016, 1546, 2709,  372, 1353,  120, 1436, 340   ],
    [  790,  2016,    0, 1084,  759, 1753, 1199, 2077,  925, 2164  ],
    [ 1627,  1546, 1084,    0, 1548, 1174,  197, 1468,  239, 1858  ],
    [   95,  2709,  759, 1548,    0, 2342, 1742, 2695, 1467, 2910  ],
    [ 2407,   372, 1753, 1173, 2342,    0,  981,  355, 1064, 711   ],
    [ 1822,  1353, 1199,  197, 1742,  981,    0, 1275,  274, 1692  ],
    [ 2759,   120, 2077, 1468, 2695,  355, 1275,    0, 1358, 460   ],
    [ 1547,  1436,  925,  239, 1467, 1064,  274, 1358,    0, 1685  ],
    [ 2941,   340, 2164, 1858, 2910,  711, 1692,  460, 1685,    0  ] )


global minimumpath
minimumpath = [
    [  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1 ],
    [  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1 ],
    [  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1 ],
    [  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1 ],
    [  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1 ],
    [  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1 ],
    [  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1 ],
    [  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1 ],
    [  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1 ],
    [  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1 ]
]

print "Original Weighted Matrix: "
print (startmatrix)
